### PR TITLE
Fix multiplayer test failures due to leaderboard load process

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Game/Database/UserLookupCache.cs [hunk 1]
```diff
@@ -27,6 +27,30 @@ public class UserLookupCache : MemoryCachingComponent<int, User>
         [ItemCanBeNull]
         public Task<User> GetUserAsync(int userId, CancellationToken token = default) => GetAsync(userId, token);
 
+        /// <summary>
+        /// Perform an API lookup on the specified users, populating a <see cref="User"/> model.
+        /// </summary>
+        /// <param name="userIds">The users to lookup.</param>
+        /// <param name="token">An optional cancellation token.</param>
+        /// <returns>.</returns>
```
Comment (smoogipoo) [inline osu.Game/Database/UserLookupCache.cs:35]: Probably should mention that this can return `null` for users that couldn't be retrieved.

**Review by smoogipoo — DISMISSED** [review]
Looks good otherwise

File: osu.Game/Database/UserLookupCache.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Database/UserLookupCache.cs:35]: Oh I meant to skip returning the nulls actually, got distracted (my usage of it doesn't guard against this either).

Comment (smoogipoo) [reply osu.Game/Database/UserLookupCache.cs:35]: Aha, yeah I think that would be even better.

Comment (peppy) [reply osu.Game/Database/UserLookupCache.cs]: Turns out doing so would be harder than expected so I just added xmldoc and checks

**Review by bdach — DISMISSED** [review]
seems okay

**Comment by peppy** [comment]
Don't merge just yet, there looks to be an issue with score updates. Potentially from the last commit.

**Comment by peppy** [comment]
Blocking as this still doesn't work.

**Review by smoogipoo — APPROVED** [review]
lgtm

**Pull request merged**